File: mobile-organizer/app/src/main/java/com/tdtuer/eventing_organizer/domain/model/Promotion.kt
package com.tdtuer.eventing_organizer.domain.model

data class Promotion(
    val id: String = "",
    val code: String = "",
    val discountValue: Double = 0.0,
    val discountType: String = "amount", // "percent" hoặc "amount"
    val description: String = "",
    val usageLimit: Int = 100,
    val usedCount: Int = 0,
    val validFrom: Long = 0L,
    val validUntil: Long = 0L,
    val minTicketQuantity: Int = 1,
    val isPublic: Boolean = false,
    val eventId: String? = null // null nếu áp dụng toàn bộ
) {
    fun isActive(): Boolean {
        val now = System.currentTimeMillis()
        return now in validFrom..validUntil && usedCount < usageLimit
    }
}

File: web/src/types/promotion.ts
/**
 * Public / organizer promotion codes returned by `/promotions`.
 */

export interface Promotion {
  id: string;
  code: string;
  name?: string;
  description?: string;
  discountType?: 'amount' | 'percent' | string;
  discountValue?: number;
  eventId?: string | null;
  validFrom?: number;
  validUntil?: number;
  usageLimit?: number;
  usedCount?: number;
  isPublic?: boolean;
  organizerId?: string;
}
